test/unit/test_utils.js

```diff
@@ -175,6 +175,52 @@ function createIdFactory(pageIndex) {
   return page.idFactory;
 }
 
+class BaseViewerMock {
```

Comment:
Sorry, but how is adding this mock class with a *completely different* (not to mention incomplete) interface than the actual `BaseViewer` class in any way helpful/useful here!?

This is actually a case where I don't think that the added unit-tests add much, if any, value, since they don't really exercise *any* of the normal code-paths in e.g. `PDFViewer`. Consequently I'd suggest removing these unit-tests from the patch.